lib/org/chromium/webidl/IDLFragmentExtractor.js

```diff
@@ -0,0 +1,95 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'IDLFragmentExtractor',
+  documentation: 'extracts IDL Fragments from HTML files',
+  requires: [
+    'foam.parsers.html.HTMLLexer'
+  ],
+
+  properties: [
+    {
+      name: 'file',
+      of: 'org.chromium.webidl.HTMLFileContent',
+      postSet: function(_, file) {
+        this.idlFragments = this.extract();
+      },
+    },
+    {
+      name: 'idlFragments',
+      // description: 'all idlFragments found in the parsed HTML file',
+    },
+  ],
+
+  methods: [
+    function extract() {
+      var self = this;
+      var lexer = self.HTMLLexer.create();
+      var OPEN = lexer.TagType.OPEN.name;
+      var CLOSE = lexer.TagType.CLOSE.name;
+      var extractAttr = function(node, attrName) {
+        var retVal;
+        node.attributes.some(function(attr) {
+          if (attr.name === attrName) {
+            retVal = attr.value.split(' ');
+          }
+        });
+        return retVal;
+      };
+
+      var results = lexer.parseString(self.file.content).value;
+
+      var idlFragments = [];
+      var tagStack = [];            // Used for tag matching.
+      var tagMatching = true;       // Set whe not inside a pre tag of interest.
+      var exclude = false;          // Set when inside example or note class.
+      var content = '';             // Used to group together related content.
+      for (var i = 0; i < results.length; i++) {
+        var item = results[i];
+        var isTag = lexer.Tag.isInstance(item);
+
+        if (!tagMatching) {
+          // Ignoring all tags. Only extracting text within pre tags.
+          if (isTag && item.nodeName === 'pre') {
```

Comment:
It is hard to say at this point whether nested `pre` tags affect the information we care about. From my current observations, there hasn't been any IDL fragments within nest `pre` tags (they seem to be mostly used for formatting). Thus, it seems like it currently is sufficient for our purposes (at least I hope so)

We could attempt to put the content through another round of processing or implement a proper HTML parser (which was my first attempt at this problem, but was scrapped since it did a lot more than it needed to and likely had other issues too).